DQM/TrackingMonitor/src/PrimaryVertexResolution.cc

```diff
@@ -394,9 +399,20 @@ void PrimaryVertexResolution::analyze(const edm::Event& iEvent, const edm::Event
   iEvent.getByToken(beamspotSrc_, hbeamspot);
   const reco::BeamSpot& beamspot = *hbeamspot;
 
-  edm::Handle<LumiScalersCollection> hscalers;
-  iEvent.getByToken(lumiScalersSrc_, hscalers);
-  const LumiScalersCollection& lumiScalers = *hscalers;
+  float lumi= -1.;
+  edm::Handle<OnlineLuminosityRecord> metaData;
+  iEvent.getByToken(metaDataSrc_, metaData);
+  if (metaData.isValid()) {
+    lumi=metaData->instLumi();
+  }
+  else{
+    edm::Handle<LumiScalersCollection> lumiScalers;
+    iEvent.getByToken(lumiScalersSrc_, lumiScalers);
```

Comment:
If the needed product is known at configuration time (as #27676 hints), it would be better to directly configure the modules to consume only the needed product.

Reply:
you mean via eras?

Reply:
In this case yes. 

Given that this pattern is copy-pasted in this PR 4 times, a possible option would be to create an additional producer that extracts the lumi value from LumiScalersCollection/OnlineLuminosityRecord after which there would be one point for the customization, and all the clients could then just read the float from the event.

Reply:
yes, that's a good suggestion. We didn't want to make era-based customizations because the list of modules needing to be treated is long (this is just the tip of the iceberg), but if the is a single event product that can be consumed for everybody that's more economical indeed.

Reply:
@arossi83, @makortel 
@mtosi (if I understood correctly) is suggesting to make use of this class: https://github.com/cms-sw/cmssw/blob/master/DQM/TrackingMonitor/src/GetLumi.cc and change the code only there (subsequently use this interface everywhere else in the client code). 
This would avoid writing an additional `EDProducer`, would this be acceptable?

Reply:
The class is certainly an improvement wrt. copy-paste. The input product customization (if that route is chosen) would still have to be repeated for each client, right?